### DISCUSSION TITLE
Build instructions

### DISCUSSION THREAD

**Discussion by giovanidiniz** [coder/code-server#3809] [Q&A] [answered]
## OS/Web Information

- Web Browser: Chrome
- Local OS: Ubuntu 20.04
- Remote OS: Ubuntu 20.04
- Remote Architecture: amd64
- `code-server --version`: 3.10.2

Hey guys,

I just learned about code-server and I’m super excited to contribute somehow.
I don’t have a ton of experience with web development, but I’m learning.

I’m trying to get up to speed and contribute so I’m trying to build code-server locally and test it. 
(I think) the build process I’m following finishes successfully, but terminal and extensions do not work properly when I run the binary.

I’m a bit puzzled about it, so I want to make sure my build process is correct.

Here is what I am doing now:

1. `yarn install && yarn build && yarn build:vscode && yarn release`
2. `cd release/ && yarn —production `
3. `cd.. && yarn release:standalone && yarn package`

At the end of these steps extract the .tar.gz file and test the binary I get a barrage of errors when I try to open a terminal or run any extensions.

I’m writing this from my phone but will add more specific errors later on.

it would be awesome if someone could verify if this is generally the proper procedure to build locally.

**Comment by jsjoeio** [comment, marked as answer]
> I just learned about code-server and I’m super excited to contribute somehow.

That's amazing! Let's see how we can help you.

Hmm...your process looks right 🤔 Let me walk through how I usually do it.

1. `yarn` to install dependencies
2. `yarn build` to build code-server
3. `yarn build:vscode` to build vscode
4. `yarn release` to create release packages
5. `cd release-packages` to find the release files
6. `tar -xzf code-server*-linux-amd64.tar.gz` to untar the release
7.  `./release-packages/code-server*-linux-amd64.tar.gz/bin/code-server`

I think that should run the binary and spin up code-server on port 8080.

Let me know if that works/helps!

**Reply by giovanidiniz** [reply]
Thanks for the info! 
Let me reproduce your specific steps and report back

**Reply by giovanidiniz** [reply]
Hi there, 
I followed your build procedure and I don’t seem to be able to replicate the issue anymore.
I must have made some mistake while running the first time around.
Thanks so much for the help

**Reply by jsjoeio** [reply]
Hooray! Sorry again for the trouble. If you have ideas or suggestions on how to improve the instructions, we're all ears (telling me hear or opening a PR). Glad it's working now!
